projects/plugins/crm/src/automation/commons/conditions/tags/class-object-tag.php

```diff
@@ -0,0 +1,261 @@
+		$tag      = $this->get_attributes()['tag'];
+
+		$this->check_for_valid_operator( $operator );
+
+		$previous_data = $data->get_previous_data();
+		$data          = $data->get_data();
+
+		if ( ! $this->check_for_valid_parameters( $operator, $data, $previous_data ) ) {
+			$this->condition_met = false;
+			return;
+		}
+
+		$this->logger->log( 'Condition: Object_Tag ' . $tag . ' => ' . $operator );
+
+		switch ( $operator ) {
+			case 'tag_added':
+				$this->condition_met = ( ! $this->has_tag_by_name( $previous_data, $tag ) && $this->has_tag_by_name( $data, $tag ) );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+			case 'tag_removed':
+				$this->condition_met = ( $this->has_tag_by_name( $previous_data, $tag ) && ! $this->has_tag_by_name( $data, $tag ) );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+			case 'has_tag':
+				$this->condition_met = $this->has_tag_by_name( $data, $tag );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+			case 'not_has_tag':
+				$this->condition_met = ! ( $this->has_tag_by_name( $data, $tag ) );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+			default:
+				$this->condition_met = false;
+				throw new Automation_Exception(
+					/* Translators: %s is the unimplemented operator. */
+					sprintf( __( 'Valid but unimplemented operator: %s', 'zero-bs-crm' ), $operator ),
+					Automation_Exception::CONDITION_OPERATOR_NOT_IMPLEMENTED
+				);
+		}
+	}
+
+	/**
+	 * Checks if the object has at least the necessary keys to detect an object
+	 * tag condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param array $data The event data.
+	 * @return bool True if the data is valid to evaluate an object tag condition, false otherwise.
+	 */
+	private function is_valid_tag_data( array $data ): bool {
+
+		if ( is_array( $data ) ) {
+			foreach ( $data as $item ) {
+				if ( ! is_array( $item ) || ! isset( $item['name'] ) ) {
+					return false;
+				}
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Get the title for the object tag condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The title.
+	 */
+	public static function get_title(): string {
+		return __( 'Object Tag', 'zero-bs-crm' );
+	}
+
+	/**
+	 * Get the slug for the object tag condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The slug 'jpcrm/condition/object_tag'.
+	 */
+	public static function get_slug(): string {
+		return 'jpcrm/condition/object_tag';
+	}
+
+	/**
+	 * Get the description for the object tag condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The description for the condition.
+	 */
+	public static function get_description(): string {
+		return __( 'Checks if an object tag matches a specified condition', 'zero-bs-crm' );
```

Comment:
"if tags of an element match..." Text for the end user. I think using the element/item word is better than entity or object.